src/action/finder/chunks.rs

```diff
@@ -0,0 +1,170 @@
+use log::warn;
+use std::fs::File;
+use std::io::{BufReader, Read, Seek, SeekFrom, Take};
+use std::path::Path;
+
+/// Implements `Iterator` trait splitting underlying `bytes` into chunks.
+#[derive(Debug)]
+pub struct Chunks<R> {
+    /// Data source for the chunks.
+    data: std::io::Bytes<R>,
+    /// Chunks reading configuration.
+    config: ChunksConfig,
+    /// Data from the previous chunk to be used as an overlap in the next chunk.
+    overlap_data: Vec<u8>,
+}
+
+pub struct GetFileChunksConfig {
+    /// Number of bytes skipped from the beginning of the file.
+    pub start_offset: u64,
+    /// Maximum number of bytes read from the file.
+    pub max_read_bytes: u64,
+    /// Desired number of bytes in chunks. Only the last chunk can be smaller
+    /// than the `bytes_per_chunk`
+    pub bytes_per_chunk: u64,
+    /// A number of bytes that the next chunk will share with the previous one.
+    pub overlap_bytes: u64,
+}
+
+pub fn get_file_chunks(
+    path: &Path,
+    config: &GetFileChunksConfig,
+) -> Option<Chunks<BufReader<Take<File>>>> {
+    let file = open_file(&path, config.start_offset, config.max_read_bytes)?;
+
+    Some(chunks(
+        BufReader::new(file),
+        ChunksConfig {
+            bytes_per_chunk: config.bytes_per_chunk,
+            overlap_bytes: config.overlap_bytes,
+        },
+    ))
+}
+
+fn open_file(
+    path: &Path,
```

Comment:
Prefer to use `AsRef<Path>` (as all standard library functions do).

Reply:
Changed to AsRef<Path>.